src/main/java/com/researchspace/dataverse/api/v1/DatasetOperations.java

```diff
@@ -69,6 +70,15 @@ public interface DatasetOperations {
 	 */
 	void uploadFile(String doi, File file);
 
+	/**
+	 * Uploads a file using a data stream.
+	 *
+	 * @param doi Identifier of the dataset that we are sending the data to.
+	 * @param file Stream of data to the contents of the file to upload.
+	 * @param filename Name of the file.
```

Comment:
If the input stream doesn't come from a file, but instead from a network or whatever, then the 'file' terminology isn't quite accurate. I think 'filename' is the name you want it to appear as in Dataverse, is that correct?